homeassistant/components/switch/pulseaudio_loopback.py

```diff
@@ -0,0 +1,136 @@
+        """Initialize the switch."""
+        self._module_id = "-1"
+        self._hass = hass
+        self._name = name
+        self._pa_host = pa_host
+        self._pa_port = int(pa_port)
+        self._sink_name = sink_name
+        self._source_name = source_name
+        self._buffer_size = int(buff_sz)
+        self._tcp_timeout = int(tcp_timeout)
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Tell the core logic if device is on."""
+        return int(self._module_id) > 0
+
+    def _send_command(self, cmd, response_expected):
+        """Send a command to the pa server using a socket."""
+        import socket
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(self._tcp_timeout)
+        try:
+            sock.connect((self._pa_host, self._pa_port))
+            _LOGGER.info("Calling pulseaudio:" + cmd)
+            sock.send((cmd + "\n").encode("utf-8"))
+            if response_expected:
+                return_data = self._get_full_response(sock)
+                _LOGGER.debug("Data received from pulseaudio: " + return_data)
+            else:
+                return_data = ""
+        finally:
+            sock.close()
+        return return_data
+
+    def _get_full_response(self, sock):
+        """Helper method to get the full response back from pulseaudio."""
+        result = ""
+        rcv_buffer = sock.recv(self._buffer_size)
+        result += rcv_buffer.decode("utf-8")
+
+        while len(rcv_buffer) == self._buffer_size:
+            rcv_buffer = sock.recv(self._buffer_size)
+            result += rcv_buffer.decode("utf-8")
+
+        return result
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        self._send_command(str.format(LOAD_CMD,
+                                      self._sink_name,
+                                      self._source_name),
+                           False)
+        self.update()
+        self.update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        self._send_command(str.format(UNLOAD_CMD, self._module_id), False)
+        self.update()
+        self.update_ha_state()
+
+    @property
+    def should_poll(self):
+        """Tell the core system whether we need polling or not (we do)."""
+        return True
+
+    def update(self):
+        """Refresh state in case an alternate process modified this data."""
+        import re
+        return_data = self._send_command("list-modules", True)
+        result = re.search(str.format(MOD_REGEX,
+                                      re.escape(self._sink_name),
+                                      re.escape(self._source_name)),
+                           return_data)
+        if result:
+            self._module_id = result.group(1)
+        else:
+            self._module_id = "-1"
```

Comment:
I had considered making it int... when interfacing with pulseaudio, I need to make it a string again, so it was just a matter of when I need to cast it.  If it was confusing to you, it will be to others too- so I'll adjust.